src/lib/operators/table_scan.hpp

```diff
@@ -60,8 +63,13 @@ class TableScan : public AbstractReadOnlyOperator {
 
   void _on_cleanup() override;
 
+  // Turns top-level uncorrelated subqueries into their value, e.g. `a = (SELECT 123)` becomes `a = 123`. This makes it
+  // easier to avoid using the more expensive ExpressionEvaluatorTableScanImpl.
+  static std::shared_ptr<AbstractExpression> _resolve_uncorrelated_subqueries(
```

Comment:
This might be a candidate for `expression_utils`, but I can't think of a second operator that might need this.